lib/sycamore/sycamore/docset.py

```diff
@@ -1531,29 +1530,45 @@ def execute(self, **kwargs) -> None:
         from sycamore.executor import Execution
         from sycamore.materialize import Materialize
 
-        if isinstance(self.plan, Materialize) and self.plan._reliability is not None:
-            mrr = self.plan._reliability
-
-            assert self.plan.child().last_node_materialize(), "first step must be materialize step"
+        if isinstance(self.plan, Materialize):
```

Comment:
I can't add the comment where I should, but in the execute docstring include basic instructions for MRR with a pointer to the full documentation in MaterializeReadReliability class.  Something like:
"""
Pipelines can be reliably executed by doing:
ctx = sycamore.init()
ctx.rewrite_rules.append(MaterializeReadReliabliity([batch_size=200]))
ds = ctx.read.<something>.<operation>...
ds.execute()

for more details see the MaterializeReadReliability class.
"""

It would be nice at some point to support reliability with take, take_all and take_streaming, but that's not urgent since nothing needs it.